### ISSUE TITLE
QAutocomplete setCurrentSelection

### ISSUE THREAD

**Issue by creeve** [quasarframework/quasar#1723]
Quasar 0.15.7

The documentation  for QAutocomplete shows setCurrentSelection as a method, but it’s actually not there anymore. It was in 0.14.x, however. Can we get it back? Or is there a new way to do the same thing? (I don’t think there is…)

**Comment by mstaack** [comment]
cant you just use `setValue` ?

**Comment by creeve** [comment]
I haven't tried, but wouldn't that just update the search text?  I want to programmatically select one of the search results.

**Comment by mstaack** [comment]
hmmm not sure.. looking at the code https://github.com/quasarframework/quasar/blob/27258b57b0b3aebe00bfd907745d2aebebff4eba/src/components/autocomplete/QAutocomplete.js#L159

it emits a `selected element` event: `this.$emit('selected', result)`

give it a try

**Comment by pdanpdan** [comment]
That one or you can use the internal `__keyboardSetCurrentSelection()` - sets the value that is shown as selected in list. But watch out that it's internal and implementation may change.

`move` and `setCurrentSelection` are not available any more, and `close` was renamed to `hide`.
For `move(offset)` there is an internal `__keyboardMoveCursor(offset)`.

**Comment by creeve** [comment]
`setValue` actually works, except for one case.  It's hard to explain, but I've made an example that shows the issue, here: https://jsfiddle.net/99949noy/35/.  Any search text will automatically select the one result and replace the search text with it's label property value.  This works unless you enter the exact label property value in the filter.  In this case, that would be the string 'One'.

Basically, I'm trying to make the search box show the 'label' field rather than the 'value' field.

The documentation needs to be updated regardless of what comes from this thread.

Thanks!

**Comment by pdanpdan** [comment]
Maybe you want something like this: https://jsfiddle.net/99949noy/85/ ?

**Comment by creeve** [comment]
Not quite like that, because I want to use `setValue` to auto-select one of the options.  Thanks for the sample though, because that helped me realize that the `value` property isn't like a key for the option and is actually just what text you want to show once the option has been selected (I know, the documentation doesn't say anything about `value` being a key, but that's what I was thinking it was).  I just had to set `value` to be the same as `label` and now it works fine.  See here if you want: https://jsfiddle.net/99949noy/97/.

Thanks a bunch!
